src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -165,4 +167,70 @@ public static Tag parseTag(String tag) throws IllegalValueException {
         }
         return tagSet;
     }
+
+    //@@author jaronchan
+    /**
+     * Parses a {@code String feature}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws IllegalValueException if the given {@code feature} is invalid.
+     */
+    public static String parseFeature(String feature) throws IllegalValueException {
+        requireNonNull(feature);
+        String trimmedFeature = feature.trim();
+        if (!(trimmedFeature.equals("details")
+                || trimmedFeature.equals("calendar")
+                || trimmedFeature.equals("scheduler"))) {
+            throw new IllegalValueException("Feature should only be either \"details\", \"calendar\" or \"scheduler\"");
+        }
+        return trimmedFeature;
+    }
+
+    /**
+     * Parses a {@code String username} into an {@code Username}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws IllegalValueException if the given {@code username} is invalid.
+     */
+    public static Username parseUsername(String username) throws IllegalValueException {
+        requireNonNull(username);
+        String trimmedUsername = username.trim();
+        if (!Username.isValidUsername(trimmedUsername)) {
```

Comment:
Hi @Esilocke, can I check, my Username class does not detect errors in the input username string when constructing. To implement your suggestion, wont I have to add a throw exception in my Username constructor? wouldn't it be weird?

Reply:
@kaisertanqr 
I wrote about that since I saw that your Username class has this line:
checkArgument(isValidUsername(username), MESSAGE_USERNAME_CONSTRAINTS);
This checkArgument method throws an IllegalArgumentException, which is an unchecked exception, and thus doesn't require any throws or try/catch blocks (and why your compiler doesn't say anything about it).

So yes, your username class does actually throw errors if you initialize it with a bad username, but because you are directly calling Username.isValidUsername here in the parser, it will never be detected. If you want to use your current implementation, I guess it's fine.